perses/rjmc/topology_proposal.py

```diff
@@ -1470,6 +1474,98 @@ def generate_oemol_from_pdb_template(pdbfile):
             atom.SetName(name_without_spaces)
         return current_oemol
 
+    @staticmethod
+    def _get_charge_difference(current_resname, new_resname):
+        """
+        return the charge of the new res - charge old res
+
+        Parameters
+        ----------
+        current_resname : str
+            three letter identifier for original residue
+        new_resnme : str
+            three letter identifier for new residue
+
+        Returns
+        -------
+        chargediff : int
+            charge(new_res) - charge(old_res)
+        """
+        assert new_resname in PolymerProposalEngine._aminos
+        assert current_resname in PolymerProposalEngine._aminos
+
+        new_rescharge, current_rescharge = 0,0
+        resname_to_charge = {current_resname: 0, new_resname: 0}
+        for resname in [new_resname, current_resname]:
+            if resname in PolymerProposalEngine._negative_aminos:
+                resname_to_charge[resname] -= 1
+            elif resname in PolymerProposalEngine._positive_aminos:
+                resname_to_charge[resname] += 1
+
+        return resname_to_charge[new_resname] - resname_to_charge[current_resname]
+
+    @staticmethod
+    def get_counterion_indices(charge_diff,
+                               old_res,
+                               new_res,
+                               new_positions,
+                               new_topology,
+                               positive_ion_name='NA',
+                               negative_ion_name='CL',
+                               radius=0.3):
+        """
+        return a randomized counterion index to neutralize (index w.r.t. new_topology)
+
+        Parameters
+        ----------
+        charge_diff : int
+            the charge difference between the new_system - old_system
+        old_res : openmm.Topology.residue
+            old mutating residue
+        new_res : openmm.Topology.residue
+            new mutating residue
+        new_positions : np.ndarray(N, 3)
```

Comment:
Are these dimensionless (and in which unit convention) or unit-bearing? Perhaps we should make them unit-bearing `Quantity` objects?

Reply:
These do not need to be `Quantity` objects, as they are just being fed into mdtraj.

Reply:
Hmm, not sure..

Reply:
If it doesn't accept `Quantity` objects, then we will see an mdtraj error, so I'm not sure that we need to change anything here